sporkfish/searcher/move_ordering/composite_heuristic.py

```diff
@@ -37,10 +48,10 @@ def evaluate(self, move: chess.Move) -> float:
         """
         # TODO: this is using is_capture twice but lets leave that for later
         # Simple aggregation for now, to be improved.
-        mvv_lva = CompositeHeuristic._MOVE_ORDER_WEIGHTS[
-            MoveOrderMode.MVV_LVA
-        ] * MvvLvaHeuristic.evaluate(self, move)
-        killer_move = CompositeHeuristic._MOVE_ORDER_WEIGHTS[
-            MoveOrderMode.KILLER_MOVE
-        ] * KillerMoveHeuristic.evaluate(self, move)
+        mvv_lva = self._config.from_dict(
```

Comment:
Ah good point, think we have indeed discussed about this before...

Sanity checking, you want something like this?
```
mo_cfg = load_config().get("MoveOrderingConfig")
_MOVE_ORDER_WEIGHTS = {
    MoveOrderMode.MVV_LVA: mo_cfg. mvv_lva_weight,
    MoveOrderMode.KILLER_MOVE: mo_cfg. killer_move_weight,
}
```

Reply:
Yep!

Reply:
so whats the benefit of creating the CompositeHeuristicConfig class and inheriting configurable since we dont pass it as an instance variable anymore, rather we have CompositeHeuristic inherit it instead and we load the weights as per @ccjeremylo suggestion?

```
class CompositeHeuristicConfig(Configurable):
    def __init__(self, mvv_lva_weight: float, killer_moves_weight: float) -> None:
        self.mvv_lva_weight = mvv_lva_weight
        self.killer_moves_weight = killer_moves_weight

```
```
class CompositeHeuristic(MvvLvaHeuristic, KillerMoveHeuristic, MoveOrderHeuristic, Configurable)
```

Reply:
@kwngggg Good point...if we take @KYLChiu's suggestion then the design is going to look a bit odd - essentially we would no longer need the `CompositeHeuristicConfig` class anymore.

I am personally in favour of your (Kev's) original design (using `CompositeHeuristicConfig`) - it is more consistent with other parts of the code, ecstatically more pleasing, and easier to read. Of course, if it turns out there is a significant perf hit, then we have no choice...

@KYLChiu you really think a small constructor with 2 lines of code (declaration and assignment) would make a significant difference?

Reply:
#127